device/transport/amqp/src/amqp_twin_client.ts

```diff
@@ -289,17 +289,17 @@ export class AmqpTwinClient extends EventEmitter {
     };
   }
 
-  private _onResponseMessage(message: Message): void {
+  private _onResponseMessage(message: AmqpMessage): void {
     debug('onResponseMessage: The downstream message is: ' + JSON.stringify(message));
     /*Codes_SRS_NODE_DEVICE_AMQP_TWIN_16_013: [The `getTwin` method shall monitor `Message` objects on the `ReceiverLink.on('message')` handler until a message with the same `correlationId` as the one that was sent is received.]*/
     /*Codes_SRS_NODE_DEVICE_AMQP_TWIN_16_021: [The `updateTwinReportedProperties` method shall monitor `Message` objects on the `ReceiverLink.on('message')` handler until a message with the same `correlationId` as the one that was sent is received.]*/
     /*Codes_SRS_NODE_DEVICE_AMQP_TWIN_16_029: [The `enableTwinDesiredPropertiesUpdates` method shall monitor `Message` objects on the `ReceiverLink.on('message')` handler until a message with the same `correlationId` as the one that was sent is received.]*/
     /*Codes_SRS_NODE_DEVICE_AMQP_TWIN_16_034: [The `disableTwinDesiredPropertiesUpdates` method shall monitor `Message` objects on the `ReceiverLink.on('message')` handler until a message with the same `correlationId` as the one that was sent is received.]*/
-    if (this._pendingTwinRequests[message.correlationId]) {
-      const pendingRequestCallback = this._pendingTwinRequests[message.correlationId];
-      delete this._pendingTwinRequests[message.correlationId];
+    if (this._pendingTwinRequests[message.properties.correlationId]) {
```

Comment:
>message.properties. [](start = 34, length = 19)

You took great care up at line 61 to determine that the message had properties before you accessed the correlationId.  No so here?

Reply:
well, we don't make the function call if we don't have one. so no need here, right?

---
In reply to: [172687346](https://github.com/Azure/azure-iot-sdk-node/pull/240#discussion_r172687346) [](ancestors = 172687346)